archivebox/extractors/readability.py

```diff
@@ -0,0 +1,83 @@
+__package__ = 'archivebox.extractors'
+
+from pathlib import Path
+
+from typing import Optional
+import json
+
+from ..index.schema import Link, ArchiveResult, ArchiveError
+from ..system import run, atomic_write
+from ..util import (
+    enforce_types,
+    download_url,
+    
+)
+from ..config import (
+    TIMEOUT,
+    SAVE_READABILITY,
+    READABILITY_BINARY,
+    READABILITY_VERSION,
+    CHROME_BINARY,
+)
+from ..logging_util import TimedProgress
+
+
+@enforce_types
+def should_save_readability(link: Link, out_dir: Optional[str]=None) -> bool:
+    out_dir = out_dir or link.link_dir
+
+    output = Path(out_dir or link.link_dir) / 'readability.json'
+    return SAVE_READABILITY and (not output.exists())
+
+
+@enforce_types
+def save_readability(link: Link, out_dir: Optional[str]=None, timeout: int=TIMEOUT) -> ArchiveResult:
+    """download reader friendly version using @mozilla/readability"""
+
+    out_dir = out_dir or link.link_dir
+    output = str(Path(out_dir).absolute() / "readability.json")
+
+    document = download_url(link.url)
```

Comment:
check for existing download from wget/singlefile/etc instead of downloading it from scratch

Reply:
I can do that, but I will leave the `download_url` call as a fallback (in case that, for example, only this method is being used)